src/audio/smart_amp_aw/aw_smart_amp_wapper.c

```diff
@@ -0,0 +1,758 @@
+#include <stdint.h>
+#include <stdlib.h>
+#include <sof/audio/smart_amp/aw_smart_amp.h>
+#include "aw_api_public.h"
+
+
+#define SOF_SMART_AMP_RX_ENABLE   	(0x10013D11)
+#define SOF_SMART_AMP_RX_PARAMS   	(0x10013D02)
+//#define SOF_SMART_AMP_TX_ENABLE   	(0x10013D13)
+#define SOF_SMART_AMP_RX_VMAX_L   	(0x10013D17)
+#define SOF_SMART_AMP_RX_VMAX_R   	(0x10013D18)
+#define SOF_SMART_AMP_RX_CALI_CFG_L (0x10013D19)
+#define SOF_SMART_AMP_RX_CALI_CFG_R (0x10013D1A)
+#define SOF_SMART_AMP_RX_RE_L       (0x10013D1B)
+#define SOF_SMART_AMP_RX_RE_R       (0x10013D1C)
+#define SOF_SMART_AMP_RX_NOISE_L    (0x10013D1D)
+#define SOF_SMART_AMP_RX_NOISE_R    (0x10013D1E)
+#define SOF_SMART_AMP_RX_F0_L       (0x10013D1F)
+#define SOF_SMART_AMP_RX_F0_R       (0x10013D20)
+#define SOF_SMART_AMP_RX_REAL_DATA_L (0x10013D21)
+#define SOF_SMART_AMP_RX_REAL_DATA_R (0x10013D22)
+#define SOF_SMART_AMP_RX_SKT_PARAMS  (0x10013D25)
+#define SOF_SMART_AMP_RX_MSG         (0x10013D2A)
+#define SOF_SMART_AMP_RX_SPIN        (0x10013D2E)
+//#define SOF_SMART_AMP_RX_SCENE       (0x10013D22)
+
+
+static int smartpa_amp_get_msg_parser(sktune_t *sktune, struct comp_dev *dev, char *data_buf, uint32_t len, int msg_id)
+{
+	uint32_t actual_data_len = len;
+	char *data_ptr = data_buf;
+	int ret;
+
+	if (sktune->sub_msg_info[msg_id].status != AW_MESG_READY) {
+		comp_err(dev, "[Awinic] set msg cmd no read, please write cmd first");
+		return -1;		
+	}
+
+	ret = sktune_api_get_data(sktune->handle, sktune->sub_msg_info[msg_id].opcode_id, data_ptr, actual_data_len);
+	if (ret < 0) {
+		comp_err(dev, "[Awinic] get msg opcode[0x%x] failed", sktune->sub_msg_info[msg_id].opcode_id);
+		sktune->sub_msg_info[msg_id].status = AW_MESG_NONE;
+		sktune->sub_msg_info[msg_id].opcode_id = 0;
+		return -1;
+	}
+	comp_info(dev, "[Awinic] get msg opcode[0x%x] done", sktune->sub_msg_info[msg_id].opcode_id);
+
+	sktune->sub_msg_info[msg_id].status = AW_MESG_NONE;
+	sktune->sub_msg_info[msg_id].opcode_id = 0;
+
+	return 0;
+}
+
+int smart_amp_get_param(sktune_t *sktune,
+			struct comp_dev *dev,
+			struct sof_ipc_ctrl_data *cdata, int max_size, uint32_t params_id)
+{
+	size_t buf_size;
+	int ret = 0;
+	int32_t *data_buf = (int32_t *)ASSUME_ALIGNED(&cdata->data->data, sizeof(uint32_t));
+
+	buf_size = cdata->num_elems;
+	
+	/* return an error in case of mismatch between num_elems and
+	 * required size
+	 */
+	if (buf_size > max_size) {
+		comp_err(dev, "[Awinic] smart_amp_get_param(): invalid num_elems %d, size %d", max_size, buf_size);
+		return -1;
+	}
+
+	switch (params_id) {
+	case SOF_SMART_AMP_RX_ENABLE : {
+		if (max_size >= sizeof(int32_t)) {
+			 data_buf[0] = sktune->enable;
+			 buf_size = sizeof(uint32_t);
+		} else {
+			comp_err(dev, "[Awinic] bad params size  %d ", max_size);
+			ret = -1;
+		}
+	} break;
+	case SOF_SMART_AMP_RX_SKT_PARAMS : {
+		ret = sktune_api_get_params(sktune->handle, (char *)data_buf, max_size);
+		if (ret < 0) {
+			comp_err(dev, "[Awinic] get params failed ");
```

Comment:
return the error? this whole function seems to fail silently to userspace on api failures